pkg/trustedresources/verify.go

```diff
@@ -91,61 +92,35 @@ func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Inter
 		return VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get matched policies: %w", err)}
 	}
 
-	tm, signature, err := prepareObjectMeta(taskObj.TaskMetadata())
-	if err != nil {
-		return VerificationResult{VerificationResultType: VerificationError, Err: err}
-	}
-	task := v1beta1.Task{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "tekton.dev/v1beta1",
-			Kind:       "Task"},
-		ObjectMeta: tm,
-		Spec:       taskObj.TaskSpec(),
-	}
-
-	return verifyResource(ctx, &task, k8s, signature, matchedPolicies)
-}
-
-// VerifyPipeline verifies the signature and public key against pipeline.
-// VerificationResult is returned with different types for different cases:
-// 1) Return VerificationResult with VerificationSkip type, when no policies are found and no-match-policy is set to ignore
-// 2) Return VerificationResult with VerificationPass type when verification passed;
-// 3) Return VerificationResult with VerificationWarn type, when no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail. Err field is filled with the warning;
-// 4) Return VerificationResult with VerificationError type when no policies are found and no-match-policy is set to fail, the resource fails to pass matched enforce verification policy, or there are errors during verification. Err is filled with the err.
-// refSource contains the source information of the pipeline.
-func VerifyPipeline(ctx context.Context, pipelineObj *v1beta1.Pipeline, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationpolicies []*v1alpha1.VerificationPolicy) VerificationResult {
-	var refSourceURI string
-	if refSource != nil {
-		refSourceURI = refSource.URI
-	}
-	matchedPolicies, err := getMatchedPolicies(pipelineObj.PipelineMetadata().Name, refSourceURI, verificationpolicies)
-	if err != nil {
-		if errors.Is(err, ErrNoMatchedPolicies) {
-			switch config.GetVerificationNoMatchPolicy(ctx) {
-			case config.IgnoreNoMatchPolicy:
-				return VerificationResult{VerificationResultType: VerificationSkip}
-			case config.WarnNoMatchPolicy:
-				logger := logging.FromContext(ctx)
-				warning := fmt.Errorf("failed to get matched policies: %w", err)
-				logger.Warnf(warning.Error())
-				return VerificationResult{VerificationResultType: VerificationWarn, Err: warning}
-			}
+	switch v := resource.(type) {
+	case *v1beta1.Task:
+		tm, signature, err := prepareObjectMeta(v.ObjectMeta)
+		if err != nil {
+			return VerificationResult{VerificationResultType: VerificationError, Err: err}
```

Comment:
can this block be moved out of the switch statement? I think `metav1.Object` should have ObjectMeta.

Can you also link the TODO for supporting v1 task/pipeline verification?